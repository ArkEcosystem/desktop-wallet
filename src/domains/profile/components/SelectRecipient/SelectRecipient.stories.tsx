import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { httpClient } from "app/services";
import React from "react";
import { StubStorage } from "tests/mocks";

import { SelectRecipient } from "./SelectRecipient";

export default { title: "Domains / Profile / Components / Select Recipient" };

const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage() });
const profile = env.profiles().create("Test profile");

export const Default = () => (
	<div className="max-w-lg space-y-8">
		<div>
			<SelectRecipient profile={profile} />
		</div>
		<div>
			<SelectRecipient profile={profile} isInvalid />
		</div>
		<div>
			<SelectRecipient profile={profile} disabled />
		</div>
		<div>
			<div className="mb-3">Selected address</div>
			<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />
		</div>
		<div>
			<div className="mb-3">Selected address (disabled)</div>
			<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" disabled />
		</div>
	</div>
);
