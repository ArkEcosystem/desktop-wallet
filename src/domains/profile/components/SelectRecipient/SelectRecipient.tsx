import { Network } from "@arkecosystem/platform-sdk/dist/coins";
import { Profile } from "@arkecosystem/platform-sdk-profiles";
import { Avatar } from "app/components/Avatar";
import { Circle } from "app/components/Circle";
import { useFormField } from "app/components/Form/useFormField";
import { Icon } from "app/components/Icon";
import { Select } from "app/components/SelectDropdown";
import cn from "classnames";
import { useProfileAddresses } from "domains/profile/hooks/use-profile-addresses";
import { SearchRecipient } from "domains/transaction/components/SearchRecipient";
import React, { useEffect, useState } from "react";

type SelectRecipientProps = {
	network?: Network;
	address?: string;
	profile: Profile;
	disabled?: boolean;
	isInvalid?: boolean;
	contactSearchTitle?: string;
	contactSearchDescription?: string;
	selectActionLabel?: string;
	onChange?: (address: string) => void;
} & Omit<React.InputHTMLAttributes<any>, "onChange">;

const ProfileAvatar = ({ address }: any) => {
	if (!address) {
		return (
			<Circle
				className="mx-3 bg-theme-secondary-200 dark:bg-theme-secondary-700 border-theme-secondary-200 dark:border-theme-secondary-700"
				size="sm"
				noShadow
			/>
		);
	}
	return <Avatar address={address} size="sm" className="mx-3" noShadow />;
};

export const SelectRecipient = React.forwardRef<HTMLInputElement, SelectRecipientProps>(
	({ address, profile, disabled, isInvalid, network, onChange }: SelectRecipientProps, ref) => {
		const [isRecipientSearchOpen, setIsRecipientSearchOpen] = useState(false);
		const [selectedAddress, setSelectedAddress] = useState(address);
		const fieldContext = useFormField();

		const isInvalidValue = isInvalid || fieldContext?.isInvalid;

		useEffect(() => {
			if (address === selectedAddress) {
				return;
			}

			setSelectedAddress(address);
		}, [address, setSelectedAddress]); // eslint-disable-line react-hooks/exhaustive-deps

		const { allAddresses } = useProfileAddresses({ profile, network });
		const recipientAddresses = allAddresses.map(({ address }) => ({
			label: address,
			value: address,
		}));

		const handleSelectAddress = (changedAddress: string) => {
			if (selectedAddress === changedAddress) {
				return;
			}

			setSelectedAddress(changedAddress);
			setIsRecipientSearchOpen(false);
			onChange?.(changedAddress);
		};

		const openRecipients = () => {
			if (disabled) {
				return;
			}

			setIsRecipientSearchOpen(true);
		};

		const onInputChange = (changedAddress: string) => {
			if (selectedAddress === changedAddress) {
				return;
			}

			setSelectedAddress(changedAddress);
			onChange?.(changedAddress);
		};

		return (
			<div>
				<div data-testid="SelectRecipient__wrapper" className="flex relative items-center w-full text-left">
					<div className="absolute left-1 z-20">
						<ProfileAvatar address={selectedAddress} />
					</div>

					<Select
						showCaret={false}
						inputClassName={cn("pl-16", { "pr-11": !isInvalidValue, "pr-18": isInvalidValue })}
						isInvalid={isInvalidValue}
						disabled={disabled}
						defaultValue={selectedAddress}
						ref={ref}
						options={recipientAddresses}
						allowFreeInput={true}
						errorClassName="mr-12"
						onChange={(option: any) => onInputChange(option.value)}
					/>

					<div
						data-testid="SelectRecipient__select-recipient"
						className="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
						onClick={openRecipients}
					>
						<Icon name="User" width={20} height={20} />
					</div>
				</div>

				<SearchRecipient
					network={network}
					isOpen={isRecipientSearchOpen}
					profile={profile}
					onAction={(address: string) => handleSelectAddress(address)}
					onClose={() => setIsRecipientSearchOpen(false)}
				/>
			</div>
		);
	},
);

SelectRecipient.displayName = "SelectRecipient";
