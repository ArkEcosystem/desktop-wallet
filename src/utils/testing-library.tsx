import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { render } from "@testing-library/react";
import { EnvironmentProvider } from "app/contexts";
import { i18n } from "app/i18n";
import { httpClient } from "app/services";
import { createMemoryHistory } from "history";
import React from "react";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import { StubStorage } from "tests/mocks";

import envFixture from "../tests/fixtures/env/data.json";

export const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage(envFixture) });

const WithProviders: React.FC = ({ children }: { children?: React.ReactNode }) => (
	<I18nextProvider i18n={i18n}>
		<EnvironmentProvider env={env}>{children}</EnvironmentProvider>
	</I18nextProvider>
);

const customRender = (component: React.ReactElement, options: any = {}) =>
	render(component, { wrapper: WithProviders, ...options });

const renderWithRouter = (
	component: React.ReactElement,
	{ routes = ["/"], history = createMemoryHistory({ initialEntries: routes }), withProviders = true } = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) =>
		withProviders ? (
			<WithProviders>
				<Router history={history}>{children}</Router>
			</WithProviders>
		) : (
			<Router history={history}>{children}</Router>
		);

	return {
		...customRender(component, { wrapper: RouterWrapper }),
		history,
	};
};

export * from "@testing-library/react";

export { customRender as render, renderWithRouter };
