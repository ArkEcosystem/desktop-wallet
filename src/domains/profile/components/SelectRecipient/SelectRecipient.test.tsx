/* eslint-disable @typescript-eslint/require-await */
import { Profile } from "@arkecosystem/platform-sdk-profiles";
import React from "react";
import { act, env, fireEvent, getDefaultProfileId, render, waitFor } from "testing-library";

import { SelectRecipient } from "./SelectRecipient";

let profile: Profile;

describe("SelectRecipient", () => {
	beforeAll(() => {
		profile = env.profiles().findById(getDefaultProfileId());
	});

	it("should render empty", () => {
		const { container } = render(<SelectRecipient profile={profile} />);
		expect(container).toMatchSnapshot();
	});

	it("should render disabled", () => {
		const { container } = render(<SelectRecipient profile={profile} disabled />);
		expect(container).toMatchSnapshot();
	});

	it("should render invalid", () => {
		const { container } = render(<SelectRecipient profile={profile} isInvalid />);
		expect(container).toMatchSnapshot();
	});

	it("should render with preselected address", () => {
		const { container } = render(<SelectRecipient profile={profile} />);
		expect(container).toMatchSnapshot();
	});

	it("should update internal state when prop changes", () => {
		const { container, rerender } = render(<SelectRecipient profile={profile} />);

		rerender(<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />);
		expect(container).toMatchSnapshot();
	});

	it("should open and close contacts modal", async () => {
		const { getByTestId } = render(
			<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		act(() => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
		});

		await waitFor(() => {
			expect(getByTestId("modal__inner")).toBeTruthy();
		});

		act(() => {
			fireEvent.click(getByTestId("modal__close-btn"));
		});

		await waitFor(() => {
			expect(() => getByTestId("modal__inner").toBeFalsy());
		});
	});

	it("should select address from contacts modal", async () => {
		const { getByTestId, getAllByTestId } = render(
			<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		act(() => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
		});

		await waitFor(() => {
			expect(getByTestId("modal__inner")).toBeTruthy();
		});

		const firstAddress = getAllByTestId("RecipientListItem__select-button")[profile.wallets().values().length];
		act(() => {
			fireEvent.click(firstAddress);
		});

		await waitFor(() => {
			expect(() => getByTestId("modal__inner").toThrow(/Unable to find an element by/));
		});

		await waitFor(() =>
			expect(getByTestId("SelectDropdownInput__input")).toHaveValue(
				profile.contacts().values()[0].addresses().values()[0].address(),
			),
		);
	});

	it("should not open contacts modal if disabled", async () => {
		const { getByTestId } = render(
			<SelectRecipient profile={profile} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" disabled />,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		act(() => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
		});

		await waitFor(() => {
			expect(() => getByTestId("modal__inner").toThrow(/Unable to find an element by/));
		});
	});

	it("should call onChange prop when entered address in input", async () => {
		const fn = jest.fn();
		const { getByTestId } = render(<SelectRecipient profile={profile} onChange={fn} />);
		const address = "bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT";
		const recipientInputField = getByTestId("SelectDropdownInput__input");

		await act(async () => {
			fireEvent.change(recipientInputField, { target: { value: address } });
		});

		expect(getByTestId("SelectDropdownInput__input")).toHaveValue(address);
		expect(fn).toBeCalledWith(address);
	});

	it("should call onChange prop if provided", async () => {
		const fn = jest.fn();

		const { getByTestId, getAllByTestId } = render(
			<SelectRecipient profile={profile} onChange={fn} address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT" />,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		act(() => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
		});

		await waitFor(() => {
			expect(getByTestId("modal__inner")).toBeTruthy();
		});

		const firstAddress = getAllByTestId("RecipientListItem__select-button")[profile.wallets().values().length];

		act(() => {
			fireEvent.click(firstAddress);
		});

		waitFor(() => expect(getByTestId("modal__inner")).toBeFalsy());

		const selectedAddressValue = profile.contacts().values()[0].addresses().values()[0].address();

		expect(getByTestId("SelectDropdownInput__input")).toHaveValue(selectedAddressValue);
		expect(fn).toBeCalledWith(selectedAddressValue);
	});

	it("should call onChange prop only when values change", async () => {
		const fn = jest.fn();

		const { getByTestId, getAllByTestId } = render(
			<SelectRecipient profile={profile} onChange={fn} address="D61mfSggzbvQgTUe6JhYKH2doHaqJ3Dyib" />,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		act(() => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
		});

		await waitFor(() => {
			expect(getByTestId("modal__inner")).toBeTruthy();
		});

		const firstAddress = getAllByTestId("RecipientListItem__select-button")[profile.wallets().values().length];

		act(() => {
			fireEvent.click(firstAddress);
		});

		waitFor(() => expect(getByTestId("modal__inner")).toBeFalsy());

		const selectedAddressValue = profile.contacts().values()[0].addresses().values()[0].address();

		expect(getByTestId("SelectDropdownInput__input")).toHaveValue(selectedAddressValue);
		expect(fn).not.toBeCalled();
	});

	it("should filter recipients list by network if provided", async () => {
		const fn = jest.fn();

		const coin = await env.coin("ARK", "ark.mainnet");

		const { getByTestId, getAllByTestId } = render(
			<SelectRecipient
				profile={profile}
				onChange={fn}
				address="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
				network={coin.network()}
			/>,
		);

		expect(() => getByTestId("modal__inner")).toThrow(/Unable to find an element by/);

		await act(async () => {
			fireEvent.click(getByTestId("SelectRecipient__select-recipient"));
			expect(() => getAllByTestId("RecipientListItem__select-button").toThrow());
		});
	});
});
