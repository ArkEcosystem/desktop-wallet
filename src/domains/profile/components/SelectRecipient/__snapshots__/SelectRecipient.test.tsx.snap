// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRecipient should render disabled 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="absolute left-1 z-20"
      >
        <div
          class="Circle-uw7h4j-0 blBpcG mx-3 bg-theme-secondary-200 dark:bg-theme-secondary-700 border-theme-secondary-200 dark:border-theme-secondary-700"
        />
      </div>
      <div
        class="relative w-full"
      >
        <input
          class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures sr-only"
          data-testid="select-list__input"
          readonly=""
          value=""
        />
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-5-menu"
            role="combobox"
          >
            <label
              for="downshift-4-input"
              id="downshift-5-label"
            />
            <div
              class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD"
              data-testid="SelectDropdownInput"
            >
              <input
                aria-autocomplete="list"
                aria-controls="downshift-5-menu"
                aria-labelledby="downshift-5-label"
                autocomplete="off"
                class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures cursor-default undefined pl-16 pr-11"
                data-testid="SelectDropdownInput__input"
                disabled=""
                id="downshift-4-input"
                placeholder="Select option"
                value=""
              />
            </div>
            <ul
              aria-labelledby="downshift-5-label"
              class="styles__SelectOptionsList-b2falt-1 cGgIup"
              id="downshift-5-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        class="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
        data-testid="SelectRecipient__select-recipient"
      >
        <div
          class="sc-bdfBwQ iyevwF"
          height="20"
          width="20"
        >
          <svg>
            user.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should render empty 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="absolute left-1 z-20"
      >
        <div
          class="Circle-uw7h4j-0 blBpcG mx-3 bg-theme-secondary-200 dark:bg-theme-secondary-700 border-theme-secondary-200 dark:border-theme-secondary-700"
        />
      </div>
      <div
        class="relative w-full"
      >
        <input
          class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures sr-only"
          data-testid="select-list__input"
          readonly=""
          value=""
        />
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-1-menu"
            role="combobox"
          >
            <label
              for="downshift-0-input"
              id="downshift-1-label"
            />
            <div
              class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD"
              data-testid="SelectDropdownInput"
            >
              <input
                aria-autocomplete="list"
                aria-controls="downshift-1-menu"
                aria-labelledby="downshift-1-label"
                autocomplete="off"
                class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures cursor-default undefined pl-16 pr-11"
                data-testid="SelectDropdownInput__input"
                id="downshift-0-input"
                placeholder="Select option"
                value=""
              />
            </div>
            <ul
              aria-labelledby="downshift-1-label"
              class="styles__SelectOptionsList-b2falt-1 cGgIup"
              id="downshift-1-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        class="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
        data-testid="SelectRecipient__select-recipient"
      >
        <div
          class="sc-bdfBwQ iyevwF"
          height="20"
          width="20"
        >
          <svg>
            user.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should render invalid 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="absolute left-1 z-20"
      >
        <div
          class="Circle-uw7h4j-0 blBpcG mx-3 bg-theme-secondary-200 dark:bg-theme-secondary-700 border-theme-secondary-200 dark:border-theme-secondary-700"
        />
      </div>
      <div
        class="relative w-full"
      >
        <input
          aria-invalid="true"
          class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures sr-only"
          data-testid="select-list__input"
          readonly=""
          value=""
        />
        <div
          class="InputGroup__InputAddonStyle-sc-1obbdqe-0 InputGroup__InputAddonEndStyle-sc-1obbdqe-2 epSJiV bWrKSO my-px mr-4"
        >
          <span
            data-testid="Input-error"
          >
            <div
              class="sc-bdfBwQ iyevwF text-theme-danger-500"
              height="20"
              width="20"
            >
              <svg>
                alert-warning.svg
              </svg>
            </div>
          </span>
        </div>
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-9-menu"
            role="combobox"
          >
            <label
              for="downshift-8-input"
              id="downshift-9-label"
            />
            <div
              class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD"
              data-testid="SelectDropdownInput"
            >
              <input
                aria-autocomplete="list"
                aria-controls="downshift-9-menu"
                aria-labelledby="downshift-9-label"
                autocomplete="off"
                class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures cursor-default  pr-16 pl-16 pr-18"
                data-testid="SelectDropdownInput__input"
                id="downshift-8-input"
                placeholder="Select option"
                value=""
              />
            </div>
            <ul
              aria-labelledby="downshift-9-label"
              class="styles__SelectOptionsList-b2falt-1 cGgIup"
              id="downshift-9-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        class="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
        data-testid="SelectRecipient__select-recipient"
      >
        <div
          class="sc-bdfBwQ iyevwF"
          height="20"
          width="20"
        >
          <svg>
            user.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should render with preselected address 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="absolute left-1 z-20"
      >
        <div
          class="Circle-uw7h4j-0 blBpcG mx-3 bg-theme-secondary-200 dark:bg-theme-secondary-700 border-theme-secondary-200 dark:border-theme-secondary-700"
        />
      </div>
      <div
        class="relative w-full"
      >
        <input
          class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures sr-only"
          data-testid="select-list__input"
          readonly=""
          value=""
        />
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-13-menu"
            role="combobox"
          >
            <label
              for="downshift-12-input"
              id="downshift-13-label"
            />
            <div
              class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD"
              data-testid="SelectDropdownInput"
            >
              <input
                aria-autocomplete="list"
                aria-controls="downshift-13-menu"
                aria-labelledby="downshift-13-label"
                autocomplete="off"
                class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures cursor-default undefined pl-16 pr-11"
                data-testid="SelectDropdownInput__input"
                id="downshift-12-input"
                placeholder="Select option"
                value=""
              />
            </div>
            <ul
              aria-labelledby="downshift-13-label"
              class="styles__SelectOptionsList-b2falt-1 cGgIup"
              id="downshift-13-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        class="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
        data-testid="SelectRecipient__select-recipient"
      >
        <div
          class="sc-bdfBwQ iyevwF"
          height="20"
          width="20"
        >
          <svg>
            user.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SelectRecipient should update internal state when prop changes 1`] = `
<div>
  <div>
    <div
      class="flex relative items-center w-full text-left"
      data-testid="SelectRecipient__wrapper"
    >
      <div
        class="absolute left-1 z-20"
      >
        <div
          class="Avatar__AvatarWrapper-sc-1vsy2s2-0 kvvaRs mx-3"
          data-testid="Avatar"
        >
          <div
            class="w-full h-full"
          >
            <img
              alt="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
              src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(255, 193, 7)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"45\\" cx=\\"20\\" cy=\\"40\\" fill=\\"rgb(244, 67, 54)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"70\\" fill=\\"rgb(233, 30, 99)\\"/><circle r=\\"55\\" cx=\\"70\\" cy=\\"50\\" fill=\\"rgb(139, 195, 74)\\"/></svg>"
              title="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
            />
          </div>
        </div>
      </div>
      <div
        class="relative w-full"
      >
        <input
          class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures sr-only"
          data-testid="select-list__input"
          readonly=""
          value="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
        />
        <div
          class="relative w-full"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="downshift-17-menu"
            role="combobox"
          >
            <label
              for="downshift-16-input"
              id="downshift-17-label"
            />
            <div
              class="InputGroup__InputGroupStyle-sc-1obbdqe-3 iqafwD"
              data-testid="SelectDropdownInput"
            >
              <input
                aria-autocomplete="list"
                aria-controls="downshift-17-menu"
                aria-labelledby="downshift-17-label"
                autocomplete="off"
                class="Input__InputStyled-wu99qs-0 eVdxgh overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-secondary-400 dark:border-theme-secondary-700 text-theme-text transition-colors duration-200 px-4 py-3 no-ligatures cursor-default undefined pl-16 pr-11"
                data-testid="SelectDropdownInput__input"
                id="downshift-16-input"
                placeholder="Select option"
                value="bP6T9GQ3kqP6T9GQ3kqP6T9GQ3kqTTTP6T9GQ3kqT"
              />
            </div>
            <ul
              aria-labelledby="downshift-17-label"
              class="styles__SelectOptionsList-b2falt-1 cGgIup"
              id="downshift-17-menu"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        class="flex absolute right-4 items-center space-x-3 cursor-pointer text-theme-primary-300 dark:text-theme-secondary-600 z-20"
        data-testid="SelectRecipient__select-recipient"
      >
        <div
          class="sc-bdfBwQ iyevwF"
          height="20"
          width="20"
        >
          <svg>
            user.svg
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
